mula/scheduler/repositories/sqlalchemy/datastore.py

```diff
@@ -55,3 +57,22 @@ def __init__(self, dsn: str) -> None:
         self.session = orm.sessionmaker(
             bind=self.engine,
         )
+
+
+def retry(max_retries=3, retry_delay=5):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            for i in range(max_retries):
+                try:
+                    return func(*args, **kwargs)
+                except Exception as e:
```

Comment:
I was wondering if it might be better to retry on a specific error (such as `sqlalchemy.exc.DBAPIError`)  rather than anything, otherwise, it could repeat calls caused by programming errors or user input.

The type(s) of exception(s) could also be an argument for the decorator, this would make the decorator even fancier.

Reply:
Yeah you're right I'll update it